src/borg/testsuite/helpers.py

```diff
@@ -65,6 +65,24 @@ def test_ssh(self, monkeypatch):
             "Location(proto='ssh', user='user', host='::', port=1234, path='/some/path', archive=None)"
         assert repr(Location('ssh://user@[::]/some/path')) == \
             "Location(proto='ssh', user='user', host='::', port=None, path='/some/path', archive=None)"
+        assert repr(Location('ssh://user@[2001:db8::]:1234/some/path::archive')) == \
+            "Location(proto='ssh', user='user', host='2001:db8::', port=1234, path='/some/path', archive='archive')"
```

Comment:
ssh does not work with brackets:
```
$ ssh [2001:db8::]
ssh: Could not resolve hostname [2001:db8::]: Temporary failure in name resolution
$ ssh 2001:db8::
ssh: connect to host 2001:db8:: port 22: Network is unreachable
```
But scp needs brackets because : is used as separator.